hack/ci/e2e.sh

```diff
@@ -23,19 +23,14 @@ REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd -P)"
 
 # our exit handler (trap)
 cleanup() {
-  # always attempt to dump logs
   kind "export" logs "${ARTIFACTS}/logs" || true
   # KIND_IS_UP is true once we: kind create
   if [[ "${KIND_IS_UP:-}" = true ]]; then
     kind delete cluster || true
   fi
-  # clean up e2e.test symlink
   rm -f _output/bin/e2e.test || true
-  # remove our tempdir
-  # NOTE: this needs to be last, or it will prevent kind delete
-  if [[ -n "${TMP_DIR:-}" ]]; then
-    rm -rf "${TMP_DIR}"
-  fi
+  # remove our tempdir, this needs to be last, or it will prevent kind delete
+  [[ -n "${TMP_DIR:-}" ]] && rm -rf "${TMP_DIR:?}"
```

Comment:
this is equivalent but shorter, except it also now explodes if we get to `rm ...` and `TMP_DIR` is not set